app/Item.php

```diff
@@ -798,4 +803,22 @@ public function getImageUrlAttribute()
     {
         return sprintf('%s%s', config('app.url'), $this->getImagePath());
     }
+
+    public function syncMatchedAuthorities(\Illuminate\Support\Collection $idsWithPivotData)
+    {
+        $idsToDetach = $this->authorities
+            ->filter(fn(Authority $authority) => $authority->pivot->automatically_matched)
+            ->pluck('id')
+            ->diff($idsWithPivotData->keys());
+
+        $this->authorities()->detach($idsToDetach);
+        $this->authorities()->syncWithoutDetaching(
+            $idsWithPivotData->map(
+                fn($pivotData) => [
+                    'automatically_matched' => true,
+                    ...$pivotData,
+                ]
+            )
+        );
+    }
```

Comment:
Just to make sure I understand: this removes all automatically_matched authorities that are not present in `$idsWithPivotData` keys?

Would this work too?

```suggestion
    {
        $this->authorities()->wherePivot('automatically_matched', true)->sync(
            $idsWithPivotData->map(
                fn($pivotData) => [
                    'automatically_matched' => true,
                    ...$pivotData,
                ]
            )
        );
    }
```

If so, could it be inlined instead? (I can only see it used once)

Reply:
This approach won't work because the `wherePivot` condition will limit the scope of pivot records, effectively leading to inserts of duplicate pivots if there already exist some with `automatically_matched = false`

Reply:
And for now this change is made only for GMUHK gallery, however it will get reused in CSV importers (or any new data source).

Reply:
@eronisko I've included your suggestion discussed via slack https://github.com/SlovakNationalGallery/webumenia.sk/pull/898/commits/84c5076013700a810f02b97e0f4d49f4fe73d474